File: frontend/hostel-frontend/src/services/roomService.js
import API from "../api/axios";

export const getRooms = () =>
  API.get("/rooms");

export const getRoomById = (roomId) =>
  API.get(`/rooms/${roomId}`);

export const createRoom = (data) =>
  API.post("/rooms", data);

export const updateRoom = (roomId,data) =>
  API.put(`/rooms/${roomId}`,data);

export const deleteRoom = (roomId) =>
  API.delete(`/rooms/${roomId}`);

export const getAvailableRooms = () =>
  API.get("/rooms/available");

export const allocateRoom = (studentId, roomNumber) =>
  API.post(`/rooms/allocate?studentId=${studentId}&roomNumber=${roomNumber}`);

export const deallocateRoom = (allocationId) =>
  API.put(`/rooms/deallocate/${allocationId}`);

export const getAllAllocations = () => {
  return API.get("/rooms/allocations");
};

export const getStudentAllocations = (studentId) =>
  API.get(`/rooms/allocations/${studentId}`);

export const getCurrentAllocation = (studentId) => {
  return API.get(`/rooms/allocations/current/${studentId}`);
};

File: backend/hostel-management/src/main/java/com/hostel/hostel_management/service/RoomService.java
package com.hostel.hostel_management.service;

import java.util.List;

import com.hostel.hostel_management.entity.Room;
import com.hostel.hostel_management.entity.RoomAllocation;

public interface RoomService {
    Room createRoom(Room room);
    List<Room> getAvailableRooms();
    RoomAllocation allocateRoom(Long studentId, String roomNumber);
    void deallocateRoom(Long allocationId);
    List<Room> getAllRooms();
    Room getRoomById(Long roomId);
    Room updateRoom(Long roomId, Room room);
    void deleteRoom(Long roomId);
    List<RoomAllocation> getStudentAllocations(Long studentId);
    RoomAllocation getCurrentAllocation(Long studentId);
    List<RoomAllocation> getAllAllocations();
   
}
